Libraries/LibGfx/JPGLoader.cpp

```diff
@@ -576,6 +599,31 @@ static bool read_reset_marker(BufferStream& stream, JPGLoadingContext& context)
     return true;
 }
 
+static bool huffman_table_reset_helper(HuffmanTableSpec& src, JPGLoadingContext& context)
+{
+    auto& table = src.type == 0 ? context.dc_tables : context.ac_tables;
+    if (src.destination_id == 0) {
+        if (table.is_empty())
+            table.append(move(src));
+        else
+            table[0] = move(src);
+    }
```

Comment:
I would `return true;` inside this block. Since you are doing a `move()` on `src`, it is logically invalid after that point, and so the code is a lot less confusing if we don't stick around longer than necessary.